calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/metadata/DocumentDbTableSchemaGenerator.java

```diff
@@ -363,41 +363,41 @@ private static void processArray(
                         : null);
                 columnMap.put(metadataColumn.getSqlName(), metadataColumn);
             }
+        }
 
-            final Map<String, String> columnNameMap = columnMap.values().stream().collect(
-                    Collectors.toMap(
-                            DocumentDbSchemaColumn::getSqlName,
-                            DocumentDbSchemaColumn::getSqlName));
-            final String indexColumnName = toName(
-                    combinePath(path, INDEX_COLUMN_NAME_PREFIX + level),
-                    columnNameMap);
-            final DocumentDbMetadataColumn indexColumn;
-            if (!columnMap.containsKey(indexColumnName)) {
-                // Add index column. Although it has no path in the original document, we will
-                // use the path of the generated index field once the original document is unwound.
-                primaryKeyColumn++;
-                indexColumn = DocumentDbMetadataColumn
-                        .builder()
-                        .sqlName(indexColumnName)
-                        .fieldPath(path)  // Once unwound, the index will be at root level so path = name.
-                        .sqlType(JdbcType.BIGINT)
-                        .isIndex(true)
-                        .isPrimaryKey(true)
-                        .index(columnMap.size() + 1)
-                        .tableName(tableName)
-                        .primaryKeyIndex(primaryKeyColumn)
-                        .foreignKeyIndex(KEY_COLUMN_NONE)
-                        .arrayIndexLevel(level)
-                        .isGenerated(true)
-                        .build();
-                columnMap.put(indexColumn.getSqlName(), indexColumn);
-            } else {
-                // Cast exception should not occur, because we are always creating DocumentDbMetadataColumn.
-                indexColumn = (DocumentDbMetadataColumn) columnMap.get(indexColumnName);
-            }
-            // Add index column to foreign keys
-            foreignKeys.add(indexColumn);
+        final Map<String, String> columnNameMap = columnMap.values().stream().collect(
```

Comment:
after changing from using path as  the key to using the table name, index was not being added to foreign key list for nested array in array or nested document in array for the case where the nested document/array in array is not in first document scanned. Moved adding index to foreign key to outside the if statement.